### DISCUSSION TITLE
Latest open-vsx Java plugins are not working in Theia

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#12610] [General] [answered]
I am using the following plugins for my theia extension:

```
  "theiaPlugins": {
    "vscode.markdown-language-features": "https://open-vsx.org/api/vscode/markdown-language-features/1.77.0/file/vscode.markdown-language-features-1.77.0.vsix",
    "vscode-builtin-extensions-pack": "https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.77.0/file/eclipse-theia.builtin-extension-pack-1.77.0.vsix",
    "redhat.java": "https://open-vsx.org/api/redhat/java/1.20.2023060704/file/redhat.java-1.20.2023060704.vsix",
    "vscjava.vscode-java-dependency": "https://open-vsx.org/api/vscjava/vscode-java-dependency/0.23.0/file/vscjava.vscode-java-dependency-0.23.0.vsix"
  },
```

With this combination, the Java Projects view is not working.  Any suggestions?

**Comment by vince-fugnitto** [comment]
@BarryDrez thank you for the discussion, are you experience the following error when opening the `java projects` view?

- https://github.com/microsoft/vscode-java-dependency/issues/763

```
Java Projects: No delegateCommandHandler for java.project.checkImportStatus
```

If so, you may need to downgrade the version as it seems to be an issue with how the extension was published to open-vsx.

**Comment by marcdumais-work** [comment]
I will make a PR on Blueprint to pin the Java Project Manager extension to a slightly older version that's known to work.

**Reply by marcdumais-work** [reply]
PR: https://github.com/eclipse-theia/theia-blueprint/pull/271

**Reply by marcdumais-work** [reply]
The PR is merged. Updated [Blueprint packages](https://projects.eclipse.org/projects/ecd.theia/downloads) should be available after CI runs (Jenkins job). The current packages are dated `2023-06-07`

**Reply by BarryDrez** [reply]
Hi @vince-fugnitto, @marcdumais-work, thank you for the update.  I am opening a workspace that has a Java project, and in the Java Projects view, it says "To view the projects, you can import the projects into the workspace".  When I try to run a class, I get the message: "Run/Debug feature requires Java language server to run in Standard mode. Do you want to switch it to Standard mode now?"  If I select Yes,  I get a message "For security concern, Java language server cannot be switched to Standard mode in untrusted workspaces.".  When I click the Manage Workspace... button, nothing happens.

**Reply by marcdumais-work** [reply]
Hi @BarryDrez ,

Are you using one of the pre-built Blueprint package we produced or do you build from sources?  Which operating system are you on? Electron version or browser?

**Reply by BarryDrez** [reply]
Hi @marcdumais-work I am building a Theia extension, and running it on Windows.  I got your last build from today, and I am using Theia 1.38 along with the plugins:  
```
    "vscode-builtin-extensions-pack": "https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.77.0/file/eclipse-theia.builtin-extension-pack-1.77.0.vsix",
    "vscjava.vscode-java-pack": "https://open-vsx.org/api/vscjava/vscode-java-pack/0.25.11/file/vscjava.vscode-java-pack-0.25.11.vsix",
    "vscjava.vscode-java-dependency": "https://open-vsx.org/api/vscjava/vscode-java-dependency/0.21.2/file/vscjava.vscode-java-dependency-0.21.2.vsix"
```

**Reply by marcdumais-work** [reply, marked as answer]
@BarryDrez Ok, I think I understand what may be happening. There's a preference, related to the vscode [Workspace Trust](https://code.visualstudio.com/docs/editor/workspace-trust) feature, with key `"security.workspace.trust.enabled"`. Can you try to set it to false, thus disabling this feature that's not yet well supported in Theia? If you want to do that same thing  programmatically, you can do it like so, in your app's `package.json`:

https://github.com/eclipse-theia/theia-blueprint/blob/master/applications/electron/package.json#L29

**Reply by marcdumais-work** [reply]
I saw [this as well](https://marketplace.visualstudio.com/items?itemName=redhat.java) that maybe can help:

![image](https://github.com/eclipse-theia/theia/assets/25749063/6b79c3a4-7373-4743-964c-6e55ab4dd843)

**Reply by BarryDrez** [reply]
Thank you @marcdumais-work !  Setting "security.workspace.trust.enabled" to false did the trick.  I can now see my workspace in the Java Projects view.  Can I set "java.server.launchMode": "Standard" in the .theia/settings file?  Will that work?

**Reply by marcdumais-work** [reply]
@BarryDrez you're welcome. Yes, that should work. You should then see the setting reflected in the `Settings` UI, under that same key (workspace level):

`.theia/settings.json:`
```json
{
    "java.compile.nullAnalysis.mode": "automatic",
    "java.server.launchMode": "Standard"
}
```

![image](https://github.com/eclipse-theia/theia/assets/25749063/0d63572c-b9ff-4ce5-b906-9343dbd6303a)

**Comment by marcdumais-work** [comment]
BTW, IIRC, there may be other edge cases that may result in the Java extension starting in "lightweight mode" (i.e. very low-features mode). I think e.g. if no JDK or JRE are found (when using the universal version of that extension). For more info, see the Extension's [README](https://github.com/redhat-developer/vscode-java#readme)

**Reply by BarryDrez** [reply]
Thanks again @marcdumais-work.  This was very helpful.
